README.md

```diff
@@ -1,20 +1,81 @@
 # Mattermost loadtest
 
-Mattermost loadtest is a standalone tool written in [Golang](https://golang.org/) for profiling [Mattermost](https://github.com/mattermost/mattermost-server) under heavy load simulating real-world usage of a server installation at scale.
+Mattermost loadtest is a standalone tool written in [Go](https://golang.org/) for profiling [Mattermost](https://github.com/mattermost/mattermost-server) under heavy load simulating real-world usage of a server installation at scale.
 
-## Goals/Features 
+## Goals/Features
 
 - No external dependencies.
 - Loosely coupled components.
-- Short, *do one thing only* functions. 
+- Short, *do one thing only* functions.
 - State handling out of main logic.
 - Theoretically no need to bulkload.
 - No need to synchronize state between multiple loadtesting instances.
 - Easy to add/remove concurrent users at execution time.
 
 ## Running
 
-`go run -v ./cmd/loadtest`
+__Pre-requisites__
+
+- Have a running Mattermost server, with known system admin credentials.
+- Copy the `config.default.json` file to `config.json` and populate the fields accordingly.
+- If you want to run a coordinator too, copy the `coordinator.default.json` file to `coordinator.json` and populate the fields accordingly.
+
+#### Populate an empty instance
+
+- `go run ./cmd/loadtest init`
+
+This populates a Mattermost instance with some teams and channels.
+
+#### Run a basic load test
+
+- `go run ./cmd/loadtest`
+
+This runs for 60 minutes and stops.
+
+#### Start a load test agent
+
+- `go run ./cmd/loadtest server`
+
+This starts a load agent in server mode, where it exposes an HTTP server to interact with the load test. Here are some commands that can be used to interact with it:
+
+- `curl -d @config/config.json http://localhost:4000/loadagent/create?id=lt1`
+
+This creates a load test with id "lt1".
+
+- `curl -X POST http://localhost:4000/loadagent/lt1/run`
+
+This starts the load test "lt1" which was created in the previous step. The load test will be started with `InitialActiveUsers` under `UsersConfiguration` in config.json.
+
+- `curl -X POST http://localhost:4000/loadagent/lt1/addusers?amount=100`
+
+This adds 100 new users to the load test.
+
+- `curl -X POST http://localhost:4000/loadagent/lt1/removeusers?amount=10`
+
+This removes 10 users from the load test.
+
+- `curl -X POST http://localhost:4000/loadagent/lt1/stop`
+
+This stops the load test.
+
+#### Start a load test coordinator
+
+- Start a load test agent.
+- Start a Prometheus server. (This is an EE feature and requires access to the enterprise codebase)
```

Comment:
so I guess the whole coordinator feature is for EE edition, right? Maybe worth mentioning it above

Reply:
done